### ISSUE TITLE
[FR] Diff text files on comparison

### ISSUE THREAD

**Issue by LeanderK** [mlflow/mlflow#1721]
It would be great if diffing of text-files such as logs or other information serialised into text-files would be supported, or more generally a way to compare output that is not just numerical metrics.

I am currently trying to compare the text-based output between different models and it's quite tedious.

**Comment by AveshCSingh** [comment]
Hi @LeanderK. Is this issue still relevant to you?

If so, did you have in mind a diff of .txt model artifacts or of params?

**Comment by LeanderK** [comment]
unfortunately I have moved on and not used mlflow in a while. I had something similar to git diff in mind. I have always stored a textual representation of all the hyper-parameter I could find for the model, implicit or explicit, so that I can compare them by comparing the files (learning rate etc.). It looked like this:

```
Adam:
- lr: 1e-4
(...)
BatchNorm:
- eps=1e-05
- momentum=0.1
- affine=True
(...)
```


 I also had a simple textual representation of the model via https://github.com/sksq96/pytorch-summary.

Being able to diff it would be great!

I imagine diffing would also be helpful for other textual outputs of the model.

**Comment by AveshCSingh** [comment]
Thanks for the reply! I agree that this would be useful.
